swap_meet/vendor.py

```diff
@@ -1,2 +1,69 @@
 class Vendor:
-    pass
\ No newline at end of file
+    def __init__(self, inventory=None):
+        if inventory:
+            self.inventory = inventory
+        else:
+            self.inventory = []
+
+
+    def add(self, added_item):
+        self.inventory.append(added_item)
+        return added_item
+
+
+    def remove(self, removed_item):
+        if removed_item not in self.inventory:
+            return False 
+        else:
+            self.inventory.remove(removed_item)
+            return removed_item  
+
+
+    def get_by_category(self, category=""):
+        items = []
+        for item in self.inventory:
+            if category == item.category:
+                items.append(item)
+        return items
+    
+
+    def swap_items(self, vendor, my_item, their_item):
+        if my_item not in self.inventory or their_item not in vendor.inventory:
+            return False
+        else:
+            self.remove(my_item)
+            vendor.add(my_item)
+            vendor.remove(their_item)
+            self.add(their_item)
+            return True
+
+
+    def swap_first_item(self, vendor):
+        if len(self.inventory) == 0 or len(vendor.inventory) == 0: 
+            return False 
+        else:
+            self.swap_items(vendor, self.inventory[0], vendor.inventory[0])
```

Comment:
Nice job reusing a method!

Here is another option if we want to consider time-space efficiency. `swap_meet` winds up being O(n) (mostly due to removing the items from the inventory). We could simply swap the items in place here to have O(1) runtime. 
```python
 self.inventory[0], vendor.inventory[0] = vendor.inventory[0], self.inventory[0]
```
More information on [swapping values with tuples](https://www.oreilly.com/library/view/python-cookbook/0596001673/ch01s02.html)